### PR TITLE
MODELIX-618 Notation language in MPS for the new text editor

### PR DESCRIPTION
**Pull request by slisson**
Several issues I discovered while working on the [modelix.text-editor](https://github.com/modelix/modelix.text-editor) project. Each commit should be reviewed individually, because the changes are mostly unrelated.

### CODE REVIEW

File: model-api-gen-runtime/src/commonMain/kotlin/org/modelix/metamodel/ITypedProperty.kt [hunk 1]
```diff
@@ -27,3 +27,28 @@ fun <ValueT> INode.setTypedPropertyValue(property: ITypedProperty<ValueT>, value
 fun <ValueT> INode.getTypedPropertyValue(property: ITypedProperty<ValueT>): ValueT {
     return property.deserializeValue(getPropertyValue(property.untyped()))
 }
+
+fun INode.getBooleanPropertyValue(property: IProperty): Boolean {
```
Comment (odzhychko) [inline model-api-gen-runtime/src/commonMain/kotlin/org/modelix/metamodel/ITypedProperty.kt:31]: nit: In public APIs like this, we wanted to document what exceptions are thrown in which circumstances.

File: mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt [hunk 2]
```diff
@@ -23,6 +23,7 @@ import jetbrains.mps.smodel.adapter.structure.link.SContainmentLinkAdapterById
 import jetbrains.mps.smodel.adapter.structure.property.SPropertyAdapterById
 import jetbrains.mps.smodel.adapter.structure.ref.SReferenceLinkAdapterById
 import org.jetbrains.mps.openapi.language.SContainmentLink
+import org.jetbrains.mps.openapi.language.SProperty
```
Comment (odzhychko) [inline mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt]: Ignoring the changed import, this commit is empty.

File: model-api-gen-runtime/src/commonMain/kotlin/org/modelix/metamodel/GeneratedConcept.kt [hunk 3]
```diff
@@ -231,7 +231,10 @@ abstract class GeneratedChildLink<ChildNodeT : ITypedNode, ChildConceptT : IConc
 
     override fun getSimpleName(): String = simpleName
 }
-fun IChildLink.typed() = this as? ITypedChildLink<ITypedNode>
+fun IChildLink.typed(): ITypedChildLink<ITypedNode> {
```
Comment (odzhychko) [inline model-api-gen-runtime/src/commonMain/kotlin/org/modelix/metamodel/GeneratedConcept.kt:234]: nit: No test for the fix for `concept.typed().untyped()`

File: mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSChangeTranslator.kt [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+import org.jetbrains.mps.openapi.module.SDependency
+import org.jetbrains.mps.openapi.module.SModule
+import org.jetbrains.mps.openapi.module.SModuleListener
+import org.jetbrains.mps.openapi.module.SModuleReference
+import org.jetbrains.mps.openapi.module.SRepository
+import org.jetbrains.mps.openapi.module.SRepositoryListener
+import org.modelix.incremental.DependencyTracking
+import org.modelix.incremental.IStateVariableReference
+
+class MPSChangeTranslator :
```
Comment (odzhychko) [inline mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSChangeTranslator.kt:38]: nit: No test for support or incremental computation in MPS-adapters.

File: mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt (hunk 2 above)
Comment (slisson) [reply mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt]: I dropped the two commits that were also fixed on main.

File: model-api-gen-runtime/src/commonMain/kotlin/org/modelix/metamodel/ITypedProperty.kt (hunk 1 above)
Comment (slisson) [reply model-api-gen-runtime/src/commonMain/kotlin/org/modelix/metamodel/ITypedProperty.kt:31]: Dropped that change.

File: mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSChangeTranslator.kt (hunk 4 above)
Comment (slisson) [reply mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSChangeTranslator.kt:38]: There are no tests for `mps-model-adapters` at all. I guess it's time to change that.

Comment (slisson) [reply mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSChangeTranslator.kt:38]: Created [MODELIX-823](https://issues.modelix.org/issue/MODELIX-823/Tests-for-mps-model-adapters) and will work on tests in a separate PR.

**Pull request merged**